lib/config/config.js

```diff
@@ -111,24 +110,50 @@ class Config {
     return this._updateConfigFile('system', modifier);
   }
 
-  // this function can preserve comments and formatting.
-  // due to fragile nature of YAWN it may not work properly in that mode.
-  // safe way is to use it in rewrite file mode
   _updateConfigFile (type, modifier) {
-    const path = this[this.configTypes[type].pathProperty];
+    const configType = this.configTypes[type];
+    const path = this[configType.pathProperty];
+
     return readFile(path, 'utf8').then(data => {
-      let text;
-      const isJSON = path.toLowerCase().endsWith('.json');
-      if (this.systemConfig && this.systemConfig.preserveStructureOnUpdates && !isJSON) {
-        const yawn = new YAWN(data);
-        yawn.json = modifier(yawn.json);
```

Comment:
I had not recalled that we introduced `preserveStructureOnUpdates`.  I thought we were still preserving it by default.  The YAML spec calls comments a "presentation detail," and most parsers/serializers don't maintain information regarding comments.  For more on this ugly wart, see here: http://yaml.org/spec/1.2/spec.html#id2766150

YAWN is the only Node.js-based YAML library that attempts to do this.  And it's a valid use case.  If we're not comfortable with the maturity of that project and we _are_ comfortable overwriting presentation details of the YAML configuration whenever Express Gateway automation touches the config files, this change is fine.  But let's explicitly draw that line in the sand before proceeding.

Reply:
Let's yank it for now.  The Go YAML parser has done some work regarding presentation, but I think we're far off from other libraries getting there.  Most of them rely on the C implementation.  Is this something that would be considered a breaking change?